accesslog.go

```diff
@@ -195,11 +215,11 @@ func (h *LoggingHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 
 	if h.logBefore {
 		writer.SetCustomLogRecord("at", "before")
-		h.logger.Log(writer.logRecord)
+		h.logger.LogContext(r.Context(), writer.logRecord)
 	}
 
 	ctx := context.WithValue(r.Context(), ctxLoggerKey, writer)
-	r = r.WithContext(ctx)
+	*r = *r.WithContext(ctx)
```

Comment:
I get it that you have to do this in https://github.com/mash/go-accesslog/pull/9/files#diff-81aabb15ee13a86a9829fa38f186effbR57 but why here ?

Reply:
This change is strictly speaking not directly related to this pull request, but from the upper middleware layer, it might be inconvenient to change the pointer of the request internally, so I changed it like this.

I will revert this change if you think it's unnecessary.

Reply:
Let's not change this line, please revert (only this line).

Reply:
reverted.